### PR TITLE
Autoload issue with ImpressionistController

### PR DESCRIPTION
**Pull request by keegnotrub**
Instead of depending on autoload to bring in the `ImpressionistController` constant, we manually bring it with a call to `require` and just move that file to `lib`.

Prior to this, Rails would emit a warning on startup about needing to use autoload durning an initializer like this, letting us know that a future version of Rails will error.

This should hopefully remove that error and still allow these controller mixins to be available.

PS. This is the recommended approach by Xavier here https://github.com/charlotte-ruby/impressionist/issues/305

### CODE REVIEW

File: impressionist.gemspec [hunk 1]
```diff
@@ -20,7 +20,7 @@ Gem::Specification.new do |s|
 
   s.add_dependency "friendly_id"
   s.add_dependency 'nokogiri', RUBY_VERSION < '2.1.0' ? '~> 1.6.0' : '~> 1'
-  s.add_dependency 'rails', '>= 3.2.15'
+  s.add_dependency 'rails', '>= 3.2.15', '< 7.1'
```
Comment (stefannibrasil) [inline impressionist.gemspec:23]: Without this constraint, we were getting an `ArgumentError` when creating a Rails 7.1 app.

File: .rubocop_todo.yml [hunk 2]
```diff
@@ -53,14 +53,14 @@ Layout/BlockAlignment:
 # Cop supports --auto-correct.
 Layout/ClosingParenthesisIndentation:
   Exclude:
```
Comment (stefannibrasil) [inline .rubocop_todo.yml:55]: All of the changes in this file are because we needed to rename the previous file to the new file inside of /lib.

**Review by christinalcole — APPROVED** [review]
I'm not actually sure why we have this gem--but the changes make sense.  If it's going to be a problem w Rails 7.1, maybe we should spend some time researching if we can drop it completely? (definitely thinking outloud/out-of-scope for this change!)

**Comment by stefannibrasil** [comment]
> I'm not actually sure why we have this gem--but the changes make sense. If it's going to be a problem w Rails 7.1, maybe we should spend some time researching if we can drop it completely? (definitely thinking outloud/out-of-scope for this change!)

I did a quick search and based on this comment from @mercedesb, we only use it for tracking the Landing pages visitors: https://convertkit.slack.com/archives/CHZGVGC2W/p1686947789442399

but it looks like this is an old question: https://convertkit.slack.com/archives/C02G2JWUX/p1578402933118100

**Review by mercedesb — APPROVED** [review]
Sneaky sneaky

**Review by jonlunsford — APPROVED** [review]
That's correct, I believe `Impressionist` is used for legacy landing pages, so usage of this should be dropping, but still in use. :|

**Pull request merged**